packages/headless/src/app/headless-engine.ts

```diff
@@ -99,9 +99,8 @@ export class HeadlessEngine<Reducers extends ReducersMapObject>
   private engine: Engine<StateFromReducersMapObject<Reducers>>;
 
   constructor(private options: HeadlessOptions<Reducers>) {
-    this.validateConfiguration(options);
-
     this.logger = buildLogger(options.loggerOptions);
+    this.validateConfiguration(options);
```

Comment:
Adding a unit test for this scenario would be useful. I'm pretty sure I inversed the order while refactoring last week. My intent was to assert that the (logger) options were valid before continuing the initialization.